src/form/Form.tsx

```diff
@@ -484,22 +490,44 @@ function Form({
     }, 750),
     [activeStep?.id, formRef, validateElements]
   );
+
+  // Debouncing the re-render due to changing values of referenced fields in hide if rules.
+  // Need to rate limit re-renders here for performance reasons.
+  const debouncedHideRuleRender = useCallback(
```

Comment:
should we just use this debounced function for all rerenders? then this doesn't have to be hide rule specific and we can unify some logic in updateFieldValues